### PR TITLE
adding rc and final release checklist templates

### PR DESCRIPTION
**Pull request by esc**
As title

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/first_rc_checklist.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+
+## llvmlite X.Y.Z
+
+* [ ] Merge to `main`.
+    - [ ] "remaining Pull-Requests from milestone".
+* [ ] Merge change log changes.
+    - [ ] "PR with changelog entries".
+* [ ] Create X.Y release branch.
+* [ ] Build conda packages on buildfarm (check "upload").
```
Comment (stuartarchibald) [inline .github/ISSUE_TEMPLATE/first_rc_checklist.md:17]:
```suggestion
* [ ] Annotated tag vX.Y.Zrc1 on release branch
* [ ] Build conda packages on buildfarm (check "upload").
```

File: .github/ISSUE_TEMPLATE/first_rc_checklist.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+* [ ] Verify packages uploaded to Anaconda Cloud and move to `numba/label/main`.
+* [ ] Build wheels (`$PYTHON_VERSIONS`) on the buildfarm.
+* [ ] Upload wheels and sdist to PyPI and verify arrival. (upload from `ci_artifacts`).
+* [ ] Initialize and verify ReadTheDocs build.
+* [ ] Send RC announcement email / post announcement to discourse group.
+* [ ] Post link to Twitter.
```
Comment (stuartarchibald) [inline .github/ISSUE_TEMPLATE/first_rc_checklist.md:23]: This section differs to what is in the equivalent in Numba and might need an update.

File: .github/ISSUE_TEMPLATE/sub_rc_checklist.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+* [ ] Verify ReadTheDocs build.
+* [ ] Send RC/FINAL announcement email / post announcement to discourse group.
+* [ ] Post link to Twitter.
+
+### Post release
+
+* [ ] Clean up `ci_artifacts` by moving files to subdirectories
+* [ ] Update release checklist template.
+* [ ] Ping Anaconda Distro team to trigger a build for `defaults` (FINAL ONLY).
+* [ ] Close milestone (and then close this release issue).
```
Comment (stuartarchibald) [inline .github/ISSUE_TEMPLATE/sub_rc_checklist.md:31]: Suggest looking at the Numba version of this and updating.

**Review by stuartarchibald — COMMENTED** [review]
Thanks for writing this up @esc. In general this looks good, but I think it would be good to include some more of the "discovered" items from Numba's equivalents docs. Thanks again.

File: .github/ISSUE_TEMPLATE/first_rc_checklist.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+---
+
+
+## llvmlite X.Y.Z
+
+* [ ] Merge to `main`.
+    - [ ] "remaining Pull-Requests from milestone".
+* [ ] Merge change log changes.
+    - [ ] "PR with changelog entries".
+* [ ] Create X.Y release branch.
```
[the remark this answers is not in the document]
Comment (esc) [reply .github/ISSUE_TEMPLATE/first_rc_checklist.md]: done in 15cb7619b6775cd259e23c611c63e9b947fff78b

File: .github/ISSUE_TEMPLATE/first_rc_checklist.md (hunk 1 above)
Comment (esc) [reply .github/ISSUE_TEMPLATE/first_rc_checklist.md:18]: done in 15cb7619b6775cd259e23c611c63e9b947fff78b

File: .github/ISSUE_TEMPLATE/first_rc_checklist.md (hunk 2 above)
Comment (esc) [reply .github/ISSUE_TEMPLATE/first_rc_checklist.md:24]: What do you mean, specifically?

File: .github/ISSUE_TEMPLATE/sub_rc_checklist.md (hunk 3 above)
Comment (esc) [reply .github/ISSUE_TEMPLATE/sub_rc_checklist.md:31]: I took a look at the Numba version and I believe this may need updating too.

Comment (esc) [reply .github/ISSUE_TEMPLATE/sub_rc_checklist.md:31]: Updates for Numba are here: https://github.com/numba/numba/pull/8118 -- will wait on review and merge of that PR and then fold the changes there into this one.

File: .github/ISSUE_TEMPLATE/first_rc_checklist.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+* [ ] Merge to `main`.
+    - [ ] "remaining Pull-Requests from milestone".
+* [ ] Merge change log changes.
+    - [ ] "PR with changelog entries".
+* [ ] Create X.Y release branch `releaseX.Y`
+* [ ] Annotated tag `vX.Y.Z` on release branch (`llvmlite` tags DO have a `v` prefix).
+* [ ] Build and upload conda packages on buildfarm (check "upload").
+* [ ] Verify packages uploaded to Anaconda Cloud and move to `numba/label/main`.
+* [ ] Build wheels and sdist on the buildfarm.
```
Comment (stuartarchibald) [inline .github/ISSUE_TEMPLATE/first_rc_checklist.md:20]:
```suggestion
* [ ] Build and upload conda packages on buildfarm (check "upload").
* [ ] Build wheels and sdist on the buildfarm (check "upload").
* [ ] Verify packages uploaded to Anaconda Cloud and move to `numba/label/main`.
```
to match Numba's equivalent. For some reason the `sub_rc_checklist.md` in Numba (and llvmlite, below, in this patch) has these the other way around. I don't think it matters way it is, but probably makes sense for them all to be consistent?

Also note the addition of "`(check "upload")` to the `Build wheels and sdist on the buildfarm.` item.

Comment (esc) [reply .github/ISSUE_TEMPLATE/first_rc_checklist.md:20]: Sounds good, better to be consistent.

**Comment by esc** [comment]
@stuartarchibald I added 4c16026a4576da1713860a7ef2b6791ad350f5c5 to this PR and opened: https://github.com/numba/numba/pull/8175 for better consistency across Numba/llvmlite and first_rc_checklist/sub_rc_checklist.

File: .github/ISSUE_TEMPLATE/sub_rc_checklist.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+
+## llvmlite X.Y.Z
+
+* [ ] Cherry-pick items from the X.Y.Z milestone into a PR.
+* [ ] Approve change log modifications and cherry-pick.
+* [ ] Merge change log modifications and cherry-picks to X.Y release branch.
+  * [ ] https://github.com/numba/numba/pull/XXXX
+* [ ] Annotated tag `vX.Y.Z` on release branch (`llvmlite` tags DO have a `v` prefix).
+* [ ] Build conda packages on buildfarm (check "upload").
```
Comment (stuartarchibald) [inline .github/ISSUE_TEMPLATE/sub_rc_checklist.md:17]:
```suggestion
* [ ] Build and upload conda packages on buildfarm (check "upload").
```

**Review by stuartarchibald — APPROVED** [review]
Thanks for the patch and fixes.

**Pull request merged**